analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -50,11 +50,12 @@ public static SymbolicConstraint ConstraintFromType(ITypeSymbol type)
             operation.Instance.ConstantValue.Value switch
             {
                 // Update DefaultValue when adding new types
-                true => SymbolicValue.True,
-                false => SymbolicValue.False,
+                true => SymbolicValue.True.WithConstraint(ObjectConstraint.NotNull),
+                false => SymbolicValue.False.WithConstraint(ObjectConstraint.NotNull),
                 null when (operation.Instance.Type ?? ConvertedType(operation.Parent)).CanBeNull() => SymbolicValue.Null,  // ToDo: MMF-2401, this will need to be reviewed
```

Comment:
I think we can remove the comment as this case does the right thing already (`CanBeNull()` returns `true` for `Nullable<T>`s)
```suggestion
                null when (operation.Instance.Type ?? ConvertedType(operation.Parent)).CanBeNull() => SymbolicValue.Null,
```

Reply:
Sound good

Reply:
To consider in the next step: Use the same trick for NotNull? It's as ugly as it sounds, but it's an option to discuss